openedx/enterprise-catalog
# Refactor update_full_content_metadata

## Description

Refactors update_full_content_metadata_task to require no arguments and work on all recently updated (within 1 hour)
ContentMetadata records.  This task is now locked based on equivalent, recently executed invocations.  It also
now retries if there are update_catalog_metadata_tasks in unready states within the past hour. ENT-4081

## Testing Instructions

* Run `./manage.py update_{full_}content_metadata` from an enterprise-catalog app shell.
* The first invocation should do some actual work.  Subsequent invocations within the same hour should emit a message about the task(s) recently having been run.
* Test the `refresh_metadata` endpoint - make a POST request to http://localhost:18160/api/v1/enterprise-catalogs/{catalog_uuid}/refresh_metadata with a body of `{"uuid": "{catalog_uuid}"}`.
* This should return a 200/OK response with a body that gives you an async task ID.
* Use the `django_celery_results_taskresult` table in `mysql-shell` to see that new tasks were started just for this catalog's query, including a task to update algolia.
* Make the request again, and you should see a REVOKED task, because we ran too recently.
* To test the retry logic - add a temporary `sleep(60)` call toward the end of the `update_catalog_metadata_task` - now submit your POST request again.  In an `app-shell`, run `./manage.py update_full_content_metadata` - this should be blocked and you should see a retry message in the worker logs - in 5 minutes the retry should occur and return successfully.

## Ticket Link

https://openedx.atlassian.net/browse/ENT-4081

## Post-review

Squash commits into discrete sets of changes
